test/e2e/provisioning/aws_e2e_test.go

```diff
@@ -0,0 +1,102 @@
+// +build e2e
+
+package provisioning
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	aws_manifest = "./testdata/machine-aws.yaml"
+)
+
+var awsScenarios = []struct {
+	// name holds short description of the test scenario, it is also used to create machines and nodes names
+	// so please don't put "strange" characters there
+	name                    string
+	osName                  string
+	containerRuntime        string
+	containerRuntimeVersion string
+	// if the -short flag was provided and this variable is set
+	// the test scenario will be run otherwise it will be skipped.
+	short bool
+}{
+	{
+		name:                    "scenario 1 Ubuntu Docker 1.13",
+		osName:                  "ubuntu",
+		containerRuntime:        "docker",
+		containerRuntimeVersion: "1.13",
+		short: true,
+	},
+
+	{
+		name:                    "scenario 2 Ubuntu Docker 17.03",
+		osName:                  "ubuntu",
+		containerRuntime:        "docker",
+		containerRuntimeVersion: "17.03",
+	},
+
+	{
+		name:                    "scenario 3 Ubuntu CRI-O 1.9",
+		osName:                  "ubuntu",
+		containerRuntime:        "cri-o",
+		containerRuntimeVersion: "1.9",
+	},
+
+	{
+		name:                    "scenario 4 CoreOS Docker 1.13",
+		osName:                  "coreos",
+		containerRuntime:        "docker",
+		containerRuntimeVersion: "1.13",
+	},
+
+	{
+		name:                    "scenario 5 CoreOS Docker 17.03",
+		osName:                  "coreos",
+		containerRuntime:        "docker",
+		containerRuntimeVersion: "17.03",
+	},
+}
+
+// TestAWSProvisioning - a test suite that exercises AWS provider
+// by requesting nodes with different combination of container runtime type, container runtime version and the OS flavour.
+func TestAWSProvisioningE2E(t *testing.T) {
+	if testing.Short() {
+		t.Parallel()
+	}
+
+	// test data
+	awsKeyID := os.Getenv("AWS_E2E_TESTS_KEY_ID")
+	awsSecret := os.Getenv("AWS_E2E_TESTS_SECRET")
+	if len(awsKeyID) == 0 || len(awsSecret) == 0 {
+		t.Fatal("unable to run the test suite, AWS_E2E_TESTS_KEY_ID or AWS_E2E_TESTS_SECRET environment variables cannot be empty")
+	}
+
+	// act
+	for _, scenario := range awsScenarios {
+		scenario := scenario // capture range variable
+		t.Run(scenario.name, func(t *testing.T) {
+			if testing.Short() == true && scenario.short == false {
+				t.SkipNow()
+			}
+			t.Parallel()
+
+			nameSufix := strings.Replace(scenario.name, " ", "-", -1)
+			nameSufix = strings.ToLower(nameSufix)
+			nameSufix = fmt.Sprintf("aws-%s", nameSufix)
+			machineName := fmt.Sprintf("machine-%s", nameSufix)
+			nodeName := fmt.Sprintf("node-%s", nameSufix)
+			params := fmt.Sprintf("<< AWS_ACCESS_KEY_ID >>=%s,<< AWS_SECRET_ACCESS_KEY >>=%s", awsKeyID, awsSecret)
```

Comment:
I must say that I do not like very much this string replacing and would prefer it if you could just do something along the lines of  `-overwrite=spec.providerconfig.accessKeyID.Value=MYKEY`. Maybe now is a good time?

Reply:
I agree it doesn't look pretty :( Those are the parameters that are passed to the `verify` tool. The `verify` tool can replace arbitrary fields in the manifest based on the input. 

We can refactor this behaviour but I'd rather do this in a separate pull request. How about that ?

Reply:
I would like to get rid of the verify tool and move it completely into the tests suite.
It feels weird having a go-native test suite which does a exec to another tool, which does the actual testing

Reply:
But a separate PR seem a good place

Reply:
I was about to suggest to get rid of the verify tool as well, so I agree there :) But its totally fine to do that in a dedicated PR

Reply:
sgtm :)